boards/arm/scobc_module1/scobc_module1_defconfig

```diff
@@ -0,0 +1,18 @@
+# SPDX-License-Identifier: Apache-2.0
+
+CONFIG_CORTEX_M_SYSTICK=y
+CONFIG_SOC_SERIES_ARM_DESIGNSTART=y
+CONFIG_SOC_ARM_DESIGNSTART_FPGA_CORTEX_M3=y
+CONFIG_BOARD_SCOBC_MODULE1=y
+CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC=48000000
+CONFIG_ARM_MPU=n
+
+CONFIG_SERIAL=y
+CONFIG_UART_INTERRUPT_DRIVEN=y
+CONFIG_CONSOLE=y
+CONFIG_UART_CONSOLE=y
+
+CONFIG_XIP=n
+CONFIG_FLASH_SIZE=0
+CONFIG_FLASH_BASE_ADDRESS=0x0
+CONFIG_BOOTLOADER_SRAM_SIZE=0
```

Comment:
Are these needed?

Reply:
Yes, these three are needed when `XIP=n`.

`CONFIG_FLASH_SIZE` is used in `zephyr/include/arch/arm/aarch32/cortex_m/scripts/linker.ld` and if you omit it, you get
```
In file included from /home/yashi/src/zephyr-project/zephyr/soc/arm/arm/designstart/linker.ld:9:
/home/yashi/src/zephyr-project/zephyr/include/arch/arm/aarch32/cortex_m/scripts/linker.ld:31:48: error: operator '==' has no left operand
   31 | #if !defined(CONFIG_XIP) && (CONFIG_FLASH_SIZE == 0)
      |                                                ^~
```

`CONFIG_FLASH_BASE_ADDRESS` and `CONFIG_BOOTLOADER_SRAM_SIZE` are needed to generate a proper `build/zephyr/misc/generated/configs.c`.  Because we have in `arch/Kconfig`

```
config FLASH_BASE_ADDRESS
	hex "Flash Base Address"
	default $(dt_chosen_reg_addr_hex,$(DT_CHOSEN_Z_FLASH)) if (XIP && (ARM || ARM64)) || !ARM
```
You won't get the value when `XIP=n`, but get
```
GEN_ABSOLUTE_SYM_KCONFIG(CONFIG_FLASH_BASE_ADDRESS, 0x);
```

With this wrong `config.c`, you get

```
/tmp/cczOtlxh.s: Assembler messages:
/tmp/cczOtlxh.s:127: Error: missing expression
```

This has been tested with
```shell
rm -rf build && west build -p -b scobc_module1 zephyr/samples/hello_world/
```